cfgov/unprocessed/apps/teachers-digital-platform/js/search-utils.js

```diff
@@ -31,6 +31,7 @@ function serializeFormFields(fields) {
     }
     return field;
   });
+  fields = fields.filter((v) => v !== 'q=');
```

Comment:
Let's make this `filter` instead of `v`

Reply:
Well I think this typo kinda proves that this variable isn't super meaningful and isn't clouding understanding of what the function is doing. I think the general advice about limiting one character variables is specifically to prevent having to hold more context in one's head. I think it's kinda the reverse situation almost with array iterators, where `v, i` is such a common convention for `currrentValue, index` that it's easier to just use them unless you're nested or have some other reason not to.

One could even imagine a more specialized function syntax, say `valueDoesntMatchFilter` that doesn't take a function as an argument, but a match string, eg, `fields = fields.valueDoesntMatchFilter('q=')` which kinda says in english what `v => v != 'q='` says in code.

Finally I think `field` actually isn't descriptive here! Something like `paramString` would be more correct, imo, this code just reassigns the value of a previous variable `fields` that used to refer to an array of objects, which then refers to an array of strings. What is a `field`? Why is it better? I'd argue it isn't!

Reply:
It's more about searching the codebase for pieces of code. Searching for `v` is potentially much noisier than searching for `field`, as it is much more likely to match parts of other words. I don't care so much what word is used, but the point is a single character is not ideal. Even `val` would be better. 

It's analogous to the search value `q` here, which we use in a few places in different contexts (prepaid cards, regs, ask, tdp, etc). Go look for references to `q` in the codebase and its usage is super obfuscated. At least in the case of JS, we don't need to do the minifiers job preemptively!

Reply:
I question the value of searching for the argument name here to this one-off utility function though, but even so, just adding in word boundaries or, because of our linting rules, parentheses allows one to target single `v` args to utility functions.

Moreover, though, I think the contextual meaning for the array methods far outweighs semantic meaning you get from actual words, kinda the same as using `i` in for loops for the index.

All this said, I've changed the logic in https://github.com/cfpb/consumerfinance.gov/pull/7839/commits/b0adce7d4d196255459e9be422d215a3e46b9726 to not have the `getSearchValues` function create the `{q: ''}` object at all so I don't have to later filter it.